pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +460,16 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	if err := f.replaceLocked(f.listLocked()); err != nil {
```

Comment:
That seems correct, although it's probably not the most efficient way to do it.
Why can't we have something like:

for _, obj := range f.listLocked() {
  if err := f.queueActionLocked(Sync, obj); err != nil {
    // handle error
  }
}